### DISCUSSION TITLE
how to record operationId instead of URL path as OperationName in skywalking agent?

### DISCUSSION THREAD

**Discussion by jeicy07** [apache/skywalking#8683] [Q&A] [answered]
The operationName in one span is now defined as **{method}:{path**}. The path part is based on OAS as shown in https://github.com/apache/skywalking/blob/master/docs/en/setup/backend/endpoint-grouping-rules.md .

However, since several paths are too long to display, is there any approach through which the agent can record the **operationId** instead of path as OperationName?

For example, can the operationName be set as "**getProducts**" in the following case instead of "**/products**"?
![截屏2022-03-15 下午6 41 12](https://user-images.githubusercontent.com/26618167/158360437-02e17dbb-0377-4b33-9fb8-9b49784ab29f.png)

**Comment by wu-sheng** [comment, marked as answer]
What is operationID? Who defines this?

**Reply by jeicy07** [reply]
The OpenAPI defines the structure as shown in the above picture. "operationId" is the method name of an API defined in user's code. For example, postMapping "/products" is implemented by the method "getProducts" and I expect the skywalking agent to record "getProducts" -- method name instead of "/products" -- path.

**Reply by wu-sheng** [reply]
No, there is no such mechanism.

**Reply by BFergerson** [reply]
It only works with SpringMVC, but you could use `USE_QUALIFIED_NAME_AS_ENDPOINT_NAME` (https://github.com/apache/skywalking-java/blob/main/apm-sniffer/apm-sdk-plugin/spring-plugins/mvc-annotation-commons/src/main/java/org/apache/skywalking/apm/plugin/spring/mvc/commons/SpringMVCPluginConfig.java#L33)

**Reply by jeicy07** [reply]
thanks！it works！
